pages/content/posts/complete-interop-features/index.md

```diff
@@ -0,0 +1,66 @@
+---
+author: "Jean-Yves Perrier"
+title: "Documenting missing interoperable web features"
+date: "2023-04-17"
+tags: ["projects"]
+ShowToc: false
+ShowBreadCrumbs: false
+excerpt: "Filling some MDN gaps by documenting missing interoperable features."
+---
+
+[MDN Web Docs](https://developer.mozilla.org/) is an invaluable information set about the web platform. With about 12000 pages, it is the most extensive corpus of documents that web developers can referene to create their websites.
+
+Unfortunately, despite being almost 20 years old, many reference pages still need to be included. We call these missing pages _gaps_, and even have tools to detect some of them:
+
+- Mozilla's [Yari](https://github.com/mdn/yari/) renderer logs missing pages as flaws when it builds a page.
+- [Dominique Hazaël-Massieux](https://www.w3.org/People/Dom/), of the [W3C](https://www.w3.org/), created the ([mdn-gaps](https://dontcallmedom.github.io/mdn-gaps/)) tool which lists all specification features with missing reference documentation. 
+
+![MDN gaps screen shot showing 5 missing fetch standard API pages and the browser support each feature](mdn-gaps-example.png)
+
+These tools enable us to identify missing content and understand the extent of the issue.
+
+## Why are there gaps on MDN?
+
+Gaps came organically. MDN Web Docs is the home of a diverse set of documents. Initially, writers put great effort into documenting Firefox technologies like XUL and XBL, and time was scarce to focus on standardized languages. While CSS, HTML, and JavaScript were well documented, docs for APIs were an afterthought. They were still living under a DOM hierarchy on MDN. With the advent of HTML as a living standard and CSS module structure initiatives (and later the EcmaScript 6 one), Mozilla Developer Network's focus shifted from proprietary to open and interoperable technologies. The shift was progressive, culminating in rebranding the site as _MDN Web Docs_, the archiving of XUL and XBL docs, and, later, the move of [Firefox Dev Tools](https://firefox-source-docs.mozilla.org/) documentation to a dedicated website.
+
+But, since the early 2010s, and even a bit earlier, standard bodies and browser makers rapidly specified and implemented new APIs. The focus was to document these new features: they were the focus of web developers, and there needed to be more good documentation about them! It was an easier move to sell internally: new technologies lead to more engagement; getting resources to document them was more straightforward than the old HTML table features, for example. The HTML5, CSS3, and ES6 initiatives allowed MDN to revamp these areas of documentation as there was an interest in positioning the web as a recognized interoperable platform.
```

Comment:
I changed certain parts of the paragraph so that the history can be conveyed more clearly. I hope it helps! 😊
```suggestion
Since the early 2010s and previous eras, standard bodies and browser makers rapidly specified and implemented new APIs. The goal was to document these new features. Additionally, these features were the focus of web developers, thus, required more good documentation. It was an easier move to sell them internally as new technologies lead to more engagement. Additionally, getting resources to document these APIs were more straightforward than for, example, the old HTML table features. The HTML5, CSS3, and ES6 initiatives allowed MDN to revamp these areas of documentation because there was an interest in positioning the web as a recognized interoperable platform.
```

Reply:
```suggestion
Since the early 2010s and in previous years, standard bodies and browser makers have rapidly specified and implemented new APIs. The goal was to document these new features. Additionally, these features were the focus of web developers, thus, required more good documentation. It was an easier move to sell them internally as new technologies led to more engagement. Additionally, getting resources to document these APIs was more straightforward than the old HTML table features. The HTML5, CSS3, and ES6 initiatives allowed MDN to revamp these areas of documentation because there was an interest in positioning the web as a recognized interoperable platform.

```